File: Simple-PostgresqlSpringDocker/src/main/java/com/pee/dockerized/postgresql/repository/UserRepository.java
package com.pee.dockerized.postgresql.repository;

import com.pee.dockerized.postgresql.entity.Role;
import com.pee.dockerized.postgresql.entity.User;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface UserRepository extends JpaRepository<User, Long> {
    Optional<User> findByEmail(String email);
    User findByRole(Role role);
}

File: src/main/kotlin/com/nuketown/kSpringDoker/repository/UserRepository.kt
package com.nuketown.kSpringDoker.repository

import com.nuketown.kSpringDoker.model.User
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.UUID

@Repository
interface UserRepository: JpaRepository<User, UUID> {
    fun findByEmail(email: String): User?
}